src/SOFe/Capital/Analytics/Top/QueryArgs.php

```diff
@@ -64,6 +66,12 @@ public function hash() : string {
         $bytes .= $this->groupingLabel;
         $bytes .= "\0";
 
+        if ($this->interval !== null) {
+            $bytes .= "\1" . Binary::writeLong($this->interval);
+        } else {
+            $bytes .= "\0";
+        }
```

Comment:
```suggestion
        $bytes .= $this->interval !== null ? "\1" . Binary::writeLong($this->interval) : "\0";
```
beautiful code:
https://github.com/SOF3/Capital/pull/36/files#diff-3184c1e50f480a62b44afe358df52a56eede3092dcf837abbbd48ddb813a6fb9R103-R105

Reply:
I don't think this code is any more beautiful. In fact, it looks much more confusing due to deeply nested expressions.

Reply:
In general, I avoid ternary operators unless it is a really short expression (such as a simple function call chain in each expression).